homeassistant/components/openevse/config_flow.py

```diff
@@ -0,0 +1,156 @@
+        return await _start_config_flow(
+            self,
+            "user",
+            user_input[CONF_NAME] if user_input else None,
+            user_input,
+            self.DEFAULTS,
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Import a config entry from configuration.yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_HOST] == import_config[CONF_HOST]:
+                _LOGGER.warning(
+                    "Already configured. This YAML configuration has already been imported. Please remove it"
+                )
+                return self.async_abort(reason="already_configured")
+
+        import_config.pop("platform", None)
+        if CONF_NAME not in import_config:
+            import_config[CONF_NAME] = DEFAULT_NAME
+
+        import_config[CONF_NAME] = slugify(import_config[CONF_NAME].lower())
+        return self.async_create_entry(
+            title=import_config[CONF_NAME], data=import_config
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry: config_entries.ConfigEntry):
+        """Return the option flow."""
+        return OpenEVSEOptionsFlow(config_entry)
+
+
+class OpenEVSEOptionsFlow(config_entries.OptionsFlow):
+    """Options flow for OpenEVSE."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        return await _start_config_flow(
+            self,
+            "init",
+            "",
+            user_input,
+            self.config_entry.data,
+            self.config_entry.entry_id,
+        )
+
+
+def _get_schema(
+    hass: HomeAssistant,
+    user_input: dict[str, Any],
+    default_dict: dict[str, Any] | None,
+    entry_id: str = None,
+) -> vol.Schema:
+    """Get a schema using the default_dict as a backup."""
+    if user_input is None:
+        user_input = {}
+
+    def _get_default(key: str, fallback_default: Any = None) -> dict[str, Any]:
+        """Get default value for key."""
+        if default_dict:
+            return user_input.get(key, default_dict.get(key, fallback_default))
+        return user_input.get(key, fallback_default)
+
+    return vol.Schema(
+        {
+            vol.Optional(
+                CONF_NAME, default=_get_default(CONF_NAME, DEFAULT_NAME)
+            ): cv.string,
+            vol.Required(
+                CONF_HOST, default=_get_default(CONF_HOST, DEFAULT_HOST)
+            ): cv.string,
+            vol.Optional(
+                CONF_USERNAME, default=_get_default(CONF_USERNAME, "")
+            ): cv.string,
+            vol.Optional(
+                CONF_PASSWORD, default=_get_default(CONF_PASSWORD, "")
+            ): cv.string,
+        },
+    )
+
+
+def _show_config_form(
```

Comment:
There is no need to create a separate function for this, it is used once and just calls another function.